src/mlpack/bindings/go/print_doc_functions_impl.hpp

```diff
@@ -0,0 +1,544 @@
+        "declaration.");
+  }
+}
+
+// Recursion base case.
+inline std::string PrintOutputOptions() { return ""; }
+
+template<typename... Args>
+std::string PrintOutputOptions(Args... args)
+{
+  // Get the list of output options for the binding.
+  std::vector<std::string> outputOptions;
+  for (auto it = CLI::Parameters().begin(); it != CLI::Parameters().end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (!d.input)
+      outputOptions.push_back(it->first);
+  }
+
+  // Now get the full list of output options that we have.
+  std::vector<std::tuple<std::string, std::string>> passedOptions;
+  GetOptions(passedOptions, args...);
+
+  // Next, iterate over all the options.
+  std::ostringstream oss;
+  for (size_t i = 0; i < outputOptions.size(); ++i)
+  {
+    // Does this option exist?
+    bool found = false;
+    size_t index = passedOptions.size();
+    for (size_t j = 0; j < passedOptions.size(); ++j)
+    {
+      if (outputOptions[i] == std::get<0>(passedOptions[j]))
+      {
+        found = true;
+        index = j;
+        break;
+      }
+    }
+
+    if (found)
+    {
+      // We have received this option, so print it.
+      if (i == 0)
+      {
+        oss << std::get<1>(passedOptions[index]);
+      }
+      else if (i > 0)
+      {
+        oss << ", ";
+        oss << std::get<1>(passedOptions[index]);
+      }
+    }
+    else
+    {
+      // We don't care about this option.
+      if (i == 0)
+      {
+        oss << "_";
+      }
+      else if (i > 0)
+      {
+        oss << ", _";
+      }
+    }
+  }
+
+  return oss.str();
+}
+
+/**
+ * Given a name of a binding and a variable number of arguments (and their
+ * contents), print the corresponding function call.
+ */
+template<typename... Args>
+std::string ProgramCall(const std::string& programName, Args... args)
+{
+  std::string result = "";
+  std::string goProgramName = CamelCase(programName, false);
+
+  // Initialize the method parameter structure
+  std::ostringstream oss;
+  oss << "    // Initialize optional parameters for " << goProgramName << "()."
+      << "\n";
+  oss << "param := mlpack." << goProgramName << "Options()\n";
+  result = oss.str();
+  oss.str(""); // Reset it.
+
+  // Now process each optional parameters.
+  oss << PrintOptionalInputs(args...) << "\n";
+  result = result + oss.str();
+  oss.str(""); // Reset it.
+
+  // Now process each output parameters.
+  oss << PrintOutputOptions(args...);
+  result = result + oss.str();
+  oss.str(""); // Reset it.
+
+  oss << " := mlpack." << goProgramName << "(";
+  result = result + oss.str();
+  oss.str(""); // Reset it.
+
+  // Now process each required input parameter.
+  oss << PrintInputOptions(args...);
+  std::string input = oss.str();
+  if (input != "")
+    result = result + input + ", ";
+  oss.str(""); // Reset it.
+  result = result + "param";
+  result = result + ")";
+  result = util::HyphenateString(result, 4);
```

Comment:
I think this is causing a problem, but I'm not sure exactly why.  The output we currently get is this:

```
    // Initialize optional parameters for Adaboost().
    param := mlpack.AdaboostOptions()
    param.Training = data
    param.Labels = labels
    param.WeakLearner = "perceptron"
    
    _, model, _, _ := mlpack.Adaboost(param)
```

but from this code we really want no indents at all:

```
// Initialize optional parameters for Adaboost().
param := mlpack.AdaboostOptions()
param.Training = data
param.Labels = labels
param.WeakLearner = "perceptron"

_, model, _, _ := mlpack.Adaboost(param)
```

However, I'm not totally sure---I haven't played with the details of it.

Reply:
Done and Thank You for the suggestion.